python/lsst/afw/geom/extent.cc

```diff
@@ -0,0 +1,228 @@
+    cls.def("setX", [](Extent<T,2> &e, T value) { e[0] = value; });
+    cls.def("setY", [](Extent<T,2> &e, T value) { e[1] = value; });
+
+    return cls;
+}
+
+// Add functionality only found in N=3 Extents
+template <typename T>
+py::class_<Extent<T,3>> declareExtent3(py::module &mod, const std::string & suffix) {
+    auto cls = declareExtent<T,3>(mod, std::string("3") + suffix);
+
+    /* Member types and enums */
+    cls.def_property_readonly_static("dimensions", [](py::object /* self */) { return 3; });
+
+    /* Constructors */
+    cls.def(py::init<int, int, int>());
+    cls.def(py::init<double, double, double>());
+
+    /* Members */
+    cls.def("getX", [](Extent<T,3> &e) { return e[0]; });
+    cls.def("getY", [](Extent<T,3> &e) { return e[1]; });
+    cls.def("getZ", [](Extent<T,3> &e) { return e[2]; });
+    cls.def("setX", [](Extent<T,3> &e, T value) { e[0] = value; });
+    cls.def("setY", [](Extent<T,3> &e, T value) { e[1] = value; });
+    cls.def("setZ", [](Extent<T,3> &e, T value) { e[2] = value; });
+
+    return cls;
+}
+
+PYBIND11_PLUGIN(_extent) {
+    py::module mod("_extent", "Python wrapper for afw _extent library");
+
+    // First declare the bases
+    auto clsExtentBase2I = declareExtentBase<int, 2>(mod, "2I");
+    auto clsExtentBase3I = declareExtentBase<int, 3>(mod, "3I");
+    auto clsExtentBase2D = declareExtentBase<double, 2>(mod, "2D");
+    auto clsExtentBase3D = declareExtentBase<double, 3>(mod, "3D");
+
+    // And then the types
+    auto clsExtent2I = declareExtent2<int>(mod, "I");
+    auto clsExtent3I = declareExtent3<int>(mod, "I");
+    auto clsExtent2D = declareExtent2<double>(mod, "D");
+    auto clsExtent3D = declareExtent3<double>(mod, "D");
+
+    /* Operators */
+
+    // Python's integer division works differently that C++'s for negative numbers - Python
+    // uses floor (rounds towards more negative), while C++ truncates (rounds towards zero).
+    // Therefore one needs to be carefull in the definition of division operators.
+    // Also note that pybind11 tries operators (like any overload) `in order'. So int has
+    // to come before double if specialization is needed.
+    clsExtent2I.def("__floordiv__", [](Extent<int,2> &e, int val) { return floor(e / static_cast<double>(val)); });
+    clsExtent3I.def("__floordiv__", [](Extent<int,3> &e, int val) { return floor(e / static_cast<double>(val)); });
+
+    clsExtent2I.def("__div__", [](Extent<int,2> &e, int val) { return floor(e / static_cast<double>(val)); });
+    clsExtent3I.def("__div__", [](Extent<int,3> &e, int val) { return floor(e / static_cast<double>(val)); });
+    clsExtent2I.def("__div__", [](Extent<int,2> &e, double val) { return e / val; });
+    clsExtent3I.def("__div__", [](Extent<int,3> &e, double val) { return e / val; });
+    clsExtent2D.def("__div__", [](Extent<double,2> &e, double val) { return e / val; });
+    clsExtent3D.def("__div__", [](Extent<double,3> &e, double val) { return e / val; });
```

Comment:
It's a bit annoying that we need to define all the `__div__` items her as clones of trudiv and floordiv.